freetextresponse/freetextresponse.py

```diff
@@ -464,40 +503,64 @@ def _get_submitted_message(self):
         Returns the message to display in the submission-received div
         """
         result = ''
-        if self.count_attempts > 0 and self._word_count_valid():
+        if self._word_count_valid():
             result = self.submitted_message
         return result
 
+    def _get_user_alert(self, ignore_attempts=False):
+        """
+        Returns the message to display in the user_alert(TBD) div
+        """
+        result = ''
+        if not self._word_count_valid():
+            result = self._get_word_count_message(ignore_attempts)
+        return result
+
     @XBlock.json_handler
     def submit(self, data, suffix=''):
         # pylint: disable=unused-argument
         """
         Processes the user's submission
         """
-        if self.max_attempts > 0 and self.count_attempts >= self.max_attempts:
-            raise StandardError(
-                _(
-                    'User has already exceeded the '
-                    'maximum number of allowed attempts'
-                )
-            )
-        self.student_answer = data['student_answer']
-        if self._word_count_valid():
-            if self.max_attempts == 0:
-                self.count_attempts = 1
-            else:
-                self.count_attempts += 1
-            self._compute_score()
+        if self.max_attempts == 0 or self.count_attempts < self.max_attempts:
```

Comment:
> I'll just add comments above the two lines in save and submit to make the relationships more clear.

@caseylitton Any time you need to add comments to clarify a boolean expression, you have an over-complicated boolean expression that should be refactored.

As you point out, this conditional appears in multiple locations.
If we're duplicating the same expression 3 times,
we benefit from a helper method, especially here since it's purely state-based
(all checks can be determined from properties on the `self` object,
no outside arguments are passed); this feels to be a natural abstraction.

If we need to fix a bug or otherwise restructure the logic behind that check,
we're better served having it centralized in a single location.
Sure, some helpers are over-kill/premature, but this PR is already an example of us needing to change this exact check.